### PR TITLE
Master nodepath for mocha

### PR DESCRIPTION
**Pull request by ericpizon**
added support for Mocha located in folder specified in NODE_PATH environment variable.

### CODE REVIEW

File: Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js [hunk 1]
```diff
@@ -80,18 +80,30 @@ function logError() {
...
-          var opt = require(mochaJsonPath);
-          if (opt && opt.path) {
-            node_modulesFolder = path.resolve(projectFolder, opt.path);
-          }
+    var node_modulesFolder = projectFolder;
+
+    var nodePath = process.env.NODE_PATH;
+    if (nodePath && nodePath.length > 0) {
+      // get mocha location from NODE_PATH env 
+      var ndex = nodePath.search(new RegExp(/node_modules$/i));
```
Comment (mjbvz) [inline Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js]: In general, don't wrap regexp literals in `new RegExp`.

But, just to be safe, make sure to use `path.sep` and a capture group to extract the path components:

```js
new RegExp('(.+?)' + path.sep + 'node_modules', 'i');
```

Comment (mjbvz) [inline Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js]: Also, I believe that NODE_PATH is actually a delimited set of paths much like PATH. This case should be handled as well. The separator is different on windows vs unix too. Here's some more info: https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders

**Comment by mjbvz** [comment]
Thanks for the PR. I've taken a look and have a few questions.

Using NODE_PATH is generally not encouraged. Can you please explain your use case for this option in a bit more detail. This will add another set of configurations that we will have to test and  maintain, so I want to make sure there is a good use case for this. 

Also, I thought that `require(...)` searches the `NODE_PATH`. Is this not the case?

**Comment by ericpizon** [comment]
Hi Matt,

Thank you for taking the time to review my pull request. You are right
about require(...) searching the node_path. We are using older version of
node.exe that does not do that. Once I switch to latest 4.x version, it
works. So, I might close this pull request and issue another one with just
a simple require('mocha') in it.

Thanks,
Eric

On Mon, Jun 6, 2016 at 12:49 PM, Matt Bierner <notifications@github.com>
wrote:

> Thanks for the PR. I've taken a look and have a few questions.
>
> Using NODE_PATH is generally not encouraged. Can you please explain your
> use case for this option in a bit more detail. This will add another set of
> configurations that we will have to test and maintain, so I want to make
> sure there is a good use case for this.
>
> Also, I thought that require(...) searches the NODE_PATH. Is this not the
> case?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/nodejstools/pull/973#issuecomment-224017339>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADjrw07diPLQMp_uLtLtxxpQA-f5ZVXIks5qJE-BgaJpZM4Ir2Qs>
> .
>



-- 
Eric E. Pizon

**Comment by mjbvz** [comment]
@ericpizon Yes, I'm much more in favor of that approach since it would also allow picking up a global mocha install: #403

**Comment by ericpizon** [comment]
Will do that.Thanks a lot!



On Tue, Jun 7, 2016 at 2:58 PM, Matt Bierner <notifications@github.com>
wrote:

> @ericpizon <https://github.com/ericpizon> Yes, I'm much more in favor of
> that approach since it would also allow picking up a global mocha install:
> #403 <https://github.com/Microsoft/nodejstools/issues/403>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/nodejstools/pull/973#issuecomment-224380021>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADjrwwAe7rudYrpHjTz-w-AAQM4KksBVks5qJb8-gaJpZM4Ir2Qs>
> .
>



-- 
Eric E. Pizon

**Comment by ericpizon** [comment]
Hi Matt,

I am looking at an erratic build issue that fails on njsproj. Is that
something I can ask you about? If not, can you direct to the right person?

Thanks,
Eric

On Tue, Jun 7, 2016 at 2:58 PM, Matt Bierner <notifications@github.com>
wrote:

> @ericpizon <https://github.com/ericpizon> Yes, I'm much more in favor of
> that approach since it would also allow picking up a global mocha install:
> #403 <https://github.com/Microsoft/nodejstools/issues/403>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/nodejstools/pull/973#issuecomment-224380021>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADjrwwAe7rudYrpHjTz-w-AAQM4KksBVks5qJb8-gaJpZM4Ir2Qs>
> .
>



-- 
Eric E. Pizon

**Comment by mjbvz** [comment]
@ericpizon Sure, I can try looking into the problem or try to link in the correct person to help out.

What problem are you seeing?

**Comment by ericpizon** [comment]
Should I use this same thread?
Thanks a lot!

On Wed, Jun 8, 2016 at 3:48 PM, Matt Bierner <notifications@github.com>
wrote:

> @ericpizon <https://github.com/ericpizon> Sure, I can try looking into
> the problem or try to link in the correct person to help out.
>
> What problem are you seeing?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/nodejstools/pull/973#issuecomment-224706523>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADjrw01NHJ6yBWIbnGfTkK8_lKlbIrDJks5qJxyfgaJpZM4Ir2Qs>
> .
>



-- 
Eric E. Pizon

**Comment by mjbvz** [comment]
@ericpizon You can but I think it would be better to open a new issue with the question so that other users can reference or find it more easily in the future.

**Comment by mjbvz** [comment]
Since it looks like you wanted to look into the other solution instead, I'm going to close this PR. Feel free to submit a new PR with that change, or let me know if you want to continue on this PR.

**Pull request closed without merging**